components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2130,20 +2130,46 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    async getStripePortalUrlForTeam(ctx: TraceContext, teamId: string): Promise<string> {
-        this.checkAndBlockUser("getStripePortalUrlForTeam");
-        const team = await this.guardTeamOperation(teamId, "update");
-        await this.ensureStripeApiIsAllowed({ team });
-        try {
-            const url = await this.stripeService.getPortalUrlForTeam(team!);
-            return url;
-        } catch (error) {
-            log.error(`Failed to get Stripe portal URL for team '${teamId}'`, error);
-            throw new ResponseError(
-                ErrorCodes.INTERNAL_SERVER_ERROR,
-                `Failed to get Stripe portal URL for team '${teamId}'`,
-            );
+    async getStripePortalUrl(ctx: TraceContext, attributionId: string): Promise<string> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
+
+        const user = this.checkAndBlockUser("getStripePortalUrl");
```

Comment:
Not really important, but shouldn't the block check be the very first call in each server method? 💭

Reply:
thanks, moved